src/__tests__/extractMeta.js

```diff
@@ -434,4 +434,109 @@ describe('Metadata extractor', () => {
             type: 'readOnly'
         });
     });
+
+    it('Extracts metadata using provided ui label', () => {
+        const props = {
+            devName: {
+                type: ElementPropTypes.string,
+                uiLabel: 'Ui label'
+            },
+            devNameTwo: {
+                type: ElementPropTypes.string,
+                uiLabel: 'Ui label two'
+            }
+        };
+
+        const extracted = extractMetadata(props);
+
+        expect(extracted).toEqual({
+            'Ui label': {
+                propName: 'devName',
+                type: 'string'
+            },
+            'Ui label two': {
+                propName: 'devNameTwo',
+                type: 'string'
+            }
+        });
+    });
+
+    it('Extracts metadata using provided ui label - mixed with non label', () => {
+        const props = {
+            devName: {
+                type: ElementPropTypes.string,
+                uiLabel: 'Ui label'
```

Comment:
@santios, Cameron and I discussed this last/this week and we came to the conclusion that `uiLabel` should be `publicLabel`. While a bit more verbose, we felt that it was a bit less ambiguous. What are your thoughts on this?

Reply:
I'm not sure about `public`. What about just `label`? No matter if we use `ui` or `public` we need to document this as neither of those are crystal clear.

Reply:
@chris-decker-volusion Additionally, `label` is used on html forms, for labels, so I think is a good name.

Reply:
We also discussed just `label`, and we're a bit split on that. I think `uiLabel` or `publicLabel` are much more clear about what the value is for. Honestly, neither of us feel very strongly about it. We're ok with leaving it as is. Just thought I'd bring it up for discussion. You are right about needing to document regardless though.

Reply:
> @chris-decker-volusion Additionally, `label` is used on html forms, for labels, so I think is a good name.

That's a fair point, and the same argument that we used to decide on key names for the oneOf ElementPropType when we were writing up documentation for the desired block contract changes. If you want to update `uiLabel` to just `label`, I'm ok with that. Cameron prefers it that way as well.

Reply:
@chris-decker-volusion Ok, updating to `label`